tests/test_sunset.py

```diff
@@ -0,0 +1,267 @@
+            "Link": '<https://new.example.com>; rel="sunset"',
+        }
+
+    def test_date_only(self, sunset_on: datetime) -> None:
+        """Test the headers when only a date is provided."""
+        headers = _make_headers(sunset_on, None)
+
+        assert headers == {"Sunset": "Mon, 01 Jan 2024 00:00:00 GMT"}
+
+    def test_no_date(self) -> None:
+        """Test the headers when you give a `null` date regardless of the link."""
+        with pytest.raises(AttributeError):
+            _make_headers(None, None)  # type: ignore[reportArgumentType]; this is what we're testing.
+
+        with pytest.raises(AttributeError):
+            _make_headers(None, "https://new.example.com")  # type: ignore[reportArgumentType]; this is what we're testing.
+
+
+class TestSunsetEndpointMiddlewareRegistry:
+    """Test the SunsetEndpointsMiddleware registry."""
+
+    @pytest.fixture
+    def mock_middleware(self) -> SunsetEndpointsMiddleware:
+        """Return a mock middleware instance with no configurations attached."""
+        return SunsetEndpointsMiddleware(Mock(spec=ASGIApp))
+
+    @pytest.fixture
+    def mock_sunset_configuration(self) -> SunsetConfiguration:
+        """Return a mock sunset configuration."""
+        sunset_configuration = Mock(spec=SunsetConfiguration)
+        sunset_configuration.sunset_on = datetime(2024, 1, 1, tzinfo=timezone.utc)
+        sunset_configuration.alternative_url = "/api/v2/deprecated"
+        return sunset_configuration
+
+    def test_register_sunset_configuration(
+        self,
+        mock_middleware: SunsetEndpointsMiddleware,
+        mock_sunset_configuration: SunsetConfiguration,
+    ) -> None:
+        """Test that registering a sunset configuration works."""
+        mock_middleware.register_sunset_configuration("/test", mock_sunset_configuration)
+
+        assert "/test" in mock_middleware._sunset_registry
+        assert mock_middleware._sunset_registry["/test"] is mock_sunset_configuration
+
+    def test_register_duplicate_sunset_configuration(
+        self,
+        mock_middleware: SunsetEndpointsMiddleware,
+        mock_sunset_configuration: SunsetConfiguration,
+    ) -> None:
+        """Test that registering a duplicate sunset configuration raises an error."""
+        mock_middleware.register_sunset_configuration("/test", mock_sunset_configuration)
+
+        with pytest.raises(ValueError, match="Endpoint /test already has a sunset configuration."):
+            mock_middleware.register_sunset_configuration("/test", mock_sunset_configuration)
+
+    def test_register_sunset_configurations(
+        self,
+        mock_middleware: SunsetEndpointsMiddleware,
+        mock_sunset_configuration: SunsetConfiguration,
+    ) -> None:
+        """Test that registering multiple sunset configurations works."""
+        mock_middleware.register_sunset_configurations(
+            **{"/test1": mock_sunset_configuration, "/test2": mock_sunset_configuration}
```

Comment:
This relates to something I've thought about. If paths are very likely to start with `/`, is there value in `register_sunset_configurations` taking `**kwargs`?